clang/lib/Driver/Driver.cpp

```diff
@@ -5827,8 +5827,9 @@ class OffloadingActionBuilder final {
             DA.add(*DeviceWrappingAction, *TC, BoundArch, Action::OFK_SYCL);
             continue;
           }
-          if (isNVPTX && Args.hasArg(options::OPT_fsycl_embed_ir)) {
-            // When compiling for Nvidia/CUDA devices and the user requested the
+          if ((isNVPTX || isAMDGCN) &&
```

Comment:
I'm not seeing any driver tests to verify that `-fsycl-embed-ir` is performing the wrapper action.  Could one be added?